drivers/entropy/fake_entropy_native_posix.c

```diff
@@ -35,18 +36,25 @@ static int entropy_native_posix_get_entropy(const struct device *dev,
 {
 	ARG_UNUSED(dev);
 
+	unsigned int bits_ready = 0;
+	uint64_t value = 0;
+
 	while (length) {
 		/*
 		 * Note that only 1 thread (Zephyr thread or HW models), runs at
 		 * a time, therefore there is no need to use random_r()
 		 */
-		long value = nsi_host_random();
+		value = value | ((uint64_t)nsi_host_random() << bits_ready);
+		/* The host random() provides a number between 0 and 2**31-1. Bit 32 is always 0 */
+		bits_ready += 31;
 
-		size_t to_copy = MIN(length, sizeof(long int));
+		size_t to_copy = MIN(length, bits_ready / 8);
 
 		memcpy(buffer, &value, to_copy);
 		buffer += to_copy;
 		length -= to_copy;
+		value >>= to_copy * 8;
+		bits_ready -= to_copy * 8;
```

Comment:
I find this a bit complicated. Why not just copy 3 bytes at a time at every loop iteration and dump the remaining 7 bits returned by the host random? I don't think we're concerned about performance to that point here?

Reply:
Fair enough :), I did overdo it indeed.